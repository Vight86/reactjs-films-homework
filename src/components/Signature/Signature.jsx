import React from 'react';
import propTypes from 'prop-types';

const Signature = ({ name }) => (
  <span>{name}</span>
);

Signature.propTypes = {
  name: propTypes.string,
};

Signature.defaultProps = {
  name: 'Vova Pekun',
};

export default Signature;
