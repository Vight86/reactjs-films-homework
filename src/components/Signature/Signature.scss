.sign {
  position: fixed;
  bottom: 0;
  left: 0;
  padding: 1em 2em;
  background-color: azure;
  border-radius: 0.5em;
  color: #5c5c5c;
  font-size: 1.6em;
  opacity: 0.5;
  transition: opacity 0.5s;

  &:hover {
    opacity: 1;
  }
}
