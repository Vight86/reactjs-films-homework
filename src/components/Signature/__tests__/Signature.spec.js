import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Signature from '../Signature';

const setUp = (name) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Signature name={name} />);
  return renderer.getRenderOutput();
};

describe('render Signature component', () => {
  it('should be div', () => {
    const result = setUp();
    expect(result.type).toBe('div');
  });

  it('render component without props', () => {
    const result = setUp();
    expect(result).toMatchSnapshot();
  });

  it('render component with props', () => {
    const result = setUp('Vova');
    expect(result).toMatchSnapshot();
  });
});
