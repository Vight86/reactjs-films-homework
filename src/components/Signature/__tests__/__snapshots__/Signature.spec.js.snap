// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Signature component render component with props 1`] = `
<div
  className="sign"
>
  Vova
</span>
`;

exports[`render Signature component render component without props 1`] = `
<div
  className="sign"
>
  Vova Pekun
</span>
`;
